core/src/main/java/feast/core/config/FeastProperties.java

```diff
@@ -39,11 +40,12 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.boot.info.BuildProperties;
-import org.springframework.stereotype.Component;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
 
 @Getter
 @Setter
-@Component
+@Configuration
```

Comment:
Can you shine some light on this change please?

Reply:
I added a security bean definition in the class. 
Below is what java doc says for `@Configuration`.
`Indicates that a class declares one or more @Bean methods and may be processed by the Spring container to generate bean definitions and service requests for those beans at runtime.`
I'm not sure if `@Component` does the same. Java doc didn't say it explicitly. If both are same it can be reverted.